lib/cluster.js

```diff
@@ -175,19 +178,28 @@ function addListeners() {
 					request.done(new Error(message.error));
 					return;
 				}
-
-				message.metrics.forEach(registry => request.responses.push(registry));
-				request.pending--;
-
-				if (request.pending === 0) {
-					// finalize
-					requests.delete(message.requestId);
-					clearTimeout(request.errorTimeout);
-
-					const registry = AggregatorRegistry.aggregate(request.responses);
-					const promString = registry.metrics();
-					request.done(null, promString);
-				}
+				fs.readFile(message.filename, 'utf8', (err, data) => {
```

Comment:
## About IPC

The IPC congestion is because of the number of request or the payload size ?
___
Can we have it as a customizable option and keep the default behavior instead ?
So users could pick if they want file vs ipc communication and there would be no breaking change.
___
Instead of sending filename to worker, could the worker keep a permanent file (named with its pid) and use IPC only to notify the master to read the file (you could use the 1st line of the file to reconcile requestId) or use `fs.watchFile`
___
Instead of file, could using unix socket or named pipes (if available) give better perf ?


## About hashing

The master process should be used only to initiate and restart workers and aggregate metrics. So it should not need that much CPU compared to workers that handle the workload. And this will put a little more load to all the workers at constant interval.

I guess you could have bad performance when you have a lot of workers and short scrape interval then.

Thus same for this feature, could it be an option to configure prom-client with hashing either in master or distributed on workers, and by default keep the current behavior to introduce no breaking changes.